### PR TITLE
Add unit tests for all repositories, the validator class and the util classes 

### PR DESCRIPTION
**Pull request by JoschuaGoetz**
I will integrate this as soon as this pull request is reviewed.

### CODE REVIEW

File: fridgify/lib/service/user_service.dart [hunk 1]
```diff
@@ -115,11 +123,14 @@ class UserService {
     throw new FailedToFetchContentException();
   }
 
+  // TODO: Refactor this. To check for an error you threw yourself is a bad
```
Comment (JoschuaGoetz) [inline fridgify/lib/service/user_service.dart:126]: @DennisRein This should be refactored. I dont think you should throw exceptions just to catch them yourself. I would call this a bad practice. Just return what is unique and what isnt unique

File: fridgify/lib/data/content_repository.dart [hunk 2]
```diff
@@ -129,6 +136,8 @@ class ContentRepository implements Repository<Content> {
 
       logger.i('ContentRepository => UPDATED SUCCESSFUL $contents');
 
+      // TODO: This doesnt work. Well it doesnt throw an error because the
```
Comment (JoschuaGoetz) [inline fridgify/lib/data/content_repository.dart:139]: @DennisRein There is a bug here

**Comment by JoschuaGoetz** [comment]
I know that there are a lot of unused cases in most switch cases. I will remove them when all functionalities are implemented

**Comment by DennisRein** [comment]
Looks good to me.

About the two questions, I refactored the user_service. The content_repository doesn't really throw a bug. Contents is just used for the logger. The given parameter content should be the same as the response, that's the reason while we update the value with it.

Would it be possible to squash the commit history before we merge?

**Comment by JoschuaGoetz** [comment]
Well rebasing and fixing the commits was harder and weirder as expected. But i managed to do it. 
Merging now

**Pull request merged**